ZAPD/ZAudio.cpp

```diff
@@ -176,22 +187,38 @@ SampleEntry* ZAudio::ParseSampleEntry(std::vector<uint8_t> audioBank,
 
 		sample->sampleDataOffset = sampleDataOffset;
 
-		if (specialLoopSamples.find(loopOffset) != specialLoopSamples.end())
-			sample->sampleLoopOffset = loopOffset;
-
-		sample->fileName = StringHelper::Sprintf("audio/samples/sample_%08X", sampleOffset);
-
-		samples[sampleOffset] = sample;
+		//if ((loopOverrideSamples.contains(((uint64_t)sampleDataOffset << 32) | loopOffset)))
+		//{
+		//	sample->sampleLoopOffset =
+		//		loopOverrideSamples[(((uint64_t)sampleDataOffset << 32) | loopOffset)];
+		//}
+		//else
+		//{
+		//	if (sampleOffsets[bankIndex][sampleDataOffset].contains(0xFFFFFFFF))
+		//	{
+		//		std::string nameBackup = sampleOffsets[bankIndex][sampleDataOffset][0xFFFFFFFF];
+		//		sampleOffsets[bankIndex][sampleDataOffset].erase(0xFFFFFFFF);
+		//		sampleOffsets[bankIndex][sampleDataOffset][loopOffset] = nameBackup;
+		//	}
+		//	sample->sampleLoopOffset = loopOffset;
+		//}
+		sample->sampleLoopOffset = 0;
+
+
+		sample->fileName = sampleOffsets[bankIndex][sampleDataOffset];  // StringHelper::Sprintf("audio/samples/sample_%08X",
+		                                                                           // sampleOffset);
+
+		samples[(uint64_t)sampleDataOffset << 32 | 0] = sample;
 
 		return sample;
 	}
 	else
 	{
-		return samples[sampleOffset];
+		return samples[(uint64_t)sampleDataOffset << 32 | 0];
```

Comment:
But then everywhere we index it, we are just shifting the value over without adding anything else, so I'm not sure the intent here.

How does this all impact 32 bit platforms (WiiU)?